### DISCUSSION TITLE
ToS violation, yes or no?

### DISCUSSION THREAD

**Discussion by Ziocash** [Tyrrrz/DiscordChatExporter#889] [General]
I'm sorry if I maybe duplicate this thread.
I made a quick search before writing this to discuss about compliancy.

I checked Discord ToS and:

When using our services, you must comply with these terms and all applicable laws, rules, and regulations, and you must only use the services for authorized and acceptable purposes. You must also adhere to our [Community Guidelines](https://discord.com/guidelines) and [other policies](https://support.discord.com/hc/articles/4420312247575), which contain more detailed rules about your content and behavior when using Discord. Fundamentally, do not do, try to do, or encourage or help others to do any of the following:

- Don’t use the services to do harm to yourself or others. Among other things, this includes trying to gain access to another user’s account or any non-public portions of the services, infringing anyone else’s intellectual property rights or any other proprietary rights, harassing, bullying, spamming, auto-messaging, or auto-dialing people through our services.
- Don’t use the services to do harm to Discord. Among other things, this includes trying to gain access to or attacking our systems, scraping us, transmitting viruses or other malicious code to our services, abusing or defrauding us or our payment systems, copying our product or using our intellectual property without permission, and misusing our reporting or customer service mechanisms.
- Don’t use the services to do anything else that’s illegal. This includes using the services to commit any crime or infringe anyone’s intellectual property rights.

We encourage you to [report content](https://support.discord.com/hc/en-us/requests/new?ticket_form_id=360000029731) or conduct that you believe violates these restrictions.

**So technically downloading chats from Discord using Token is accepted, it is not tolerated to use the token to spam things**.

If you have further info, let me know, maybe i didn't check something or I missed anything else.

**Comment by Technetium1** [comment]
This is not *technically* ToS compliant, though an effort is made to have requests made reasonably slow enough to pass as a client.

### **_Use is at your own risk. Accounts have been shutdown - safety is not certain._**

**Comment by 96-LB** [comment]
From the [Respect Discord](https://discord.com/guidelines#respect-discord) section:

> * **Do not use self-bots or user-bots**. Each account must be associated with a human, not a bot. Self-bots put strain on Discord’s infrastructure and our ability to run our services. For more information, you can read our Developer Policies [here](https://discord.com/developers/docs/policy).

Using a token from a user account is a violation of the terms of service. Using a token from a bot account, on the other hand, is compliant and perfectly okay. It's strongly encouraged that you use a bot token with DiscordChatExporter to avoid breaking ToS.

**Reply by Snowphey** [reply]
Could someone explain how to use a bot token with DiscordChatExported? I never used bots on Discord so I have no idea how it works.

**Reply by 96-LB** [reply]
If you go [here](https://discord.com/developers/applications), you can create a new application and give it a new bot account. Doing so should provide you with a token to copy and store somewhere safe. This is the token you'll give to DiscordChatExporter—no knowledge of bot coding is required, as the program already handles all of that for you.

The token will be able to export any channels which the bot can see, so you'll have to add the bot to any servers which you want to archive. You should be able to generate a link to add your bot under OAuth2 -> Url Generator. Once on that page, click the checkbox next to "bot" to get started, and then select any relevant permissions such as "read messages/view channels" and "read message history". This link can now be used to invite the bot to any server which you have "manage server" permissions on. If you don't have that permission, you'll need to work something out with the server owner/moderators.

**Reply by Snowphey** [reply]
Thank you very much for this fast and detailed answer!

**Reply by nonami1612** [reply]
Is there a chance that this will take down my server and/or my account? Sorry if this is a ridiculous question, I can't help but worry.

**Reply by Ziocash** [reply]
> Is there a chance that this will take down my server and/or my account? Sorry if this is a ridiculous question, I can't help but worry.

Yes, using a bot token is more secure

**Reply by 96-LB** [reply]
@nonami1612 I believe that if your account is terminated then any servers you own are taken down with it. You might be able to appeal a ban, but you're strongly recommended to comply with the terms of service and use a bot account, which has a 0% chance of getting you banned. I've heard very few reports of anybody losing their account because they used DiscordChatExporter with a user token, but it can happen.

**Reply by nonami1612** [reply]
@96-LB Okay, thank you. I've just been banned before for no reason (it was repealed thankfully), so I can't help but worry.

**Comment by Ziocash** [comment]
I opened this discussion on purpose, just to explain to Discord that not all "self-bots" are intended for damaging their infrastructure.
I also hope one day that the answer can be "no" instead of "yes".

**Comment by KeqiangSun** [comment]
In the latest [Discord Developer Policy](https://discord.com/developers/docs/policies-and-agreements/developer-policy), there is a term that reads:
> You may not mine or scrape any data, content, or information available on or through Discord services. 
>
I noticed in other strict terms, they said "Do not", rather than "You may not", like:
> Do not use API Data for any purpose outside of what is necessary to provide your stated functionality.
>
There might be some difference between them.
Anyone knows to interpolate that correctly? Are we officially allowed to export history from Discord?
Thank you!

**Comment by stevenOCR** [comment]
I just got dinged this week for Platform Manipulation disabling my account's ability to send chats for two days. I've been suing DiscordChatExporter a few times this week each of which go back about two months of light to medium chats frequency. So yes you can definitely lose your account's good standing for overusing this tool.

**Comment by YourHabibiAmir** [comment]
Does this put my data at risk? Like the actual texts have sensitive data

**Comment by EclipsedStar0** [comment]
Was exporting a server's chat logs with the respect api rate limits and got an email saying 'You broke Discord's Community Guidelines'. You can DEFINITELY get violations for 'Platform Manipulation'. Especially if you export multiple things at once, you can rack up multiple community violations in one go.
